runtime/parser/expression.go

```diff
@@ -198,7 +199,7 @@ func defineExpr(def any) {
 func setExprNullDenotation(tokenType lexer.TokenType, nullDenotation exprNullDenotationFunc) {
 	current := exprNullDenotations[tokenType]
 	if current != nil {
-		panic(NewUnpositionedSyntaxError(
+		panic(fmt.Errorf(
```

Comment:
Maybe panic with an `errors.UnexpectedError` here and in other similar places.
```suggestion
		panic(errors.NewUnexpectedError(
```